plugins/ccgx/fu-ccgx-hid-device.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2020 Cypress Semiconductor Corporation.
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-ccgx-hid-device.h"
+
+struct _FuCcgxHidDevice
+{
+	FuHidDevice		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuCcgxHidDevice, fu_ccgx_hid_device, FU_TYPE_HID_DEVICE)
+
+#define FU_CCGX_HID_DEVICE_TIMEOUT	5000 /* ms */
+
+static gboolean
+fu_ccgx_hid_device_detach (FuDevice *device, GError **error)
+{
+	guint8 buf[5] = {0xEE, 0xBC, 0xA6, 0xB9, 0xA8};
+	if (!fu_hid_device_set_report (FU_HID_DEVICE (device), buf[0],
+				       buf, sizeof(buf),
+				       FU_CCGX_HID_DEVICE_TIMEOUT,
+				       FU_HID_DEVICE_FLAG_NONE,
+				       error)) {
+		g_prefix_error (error, "mfg mode error: ");
+		return FALSE;
+	}
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
+	fu_device_add_flag (device, FWUPD_DEVICE_FLAG_WILL_DISAPPEAR);
+	return TRUE;
+}
+
+static void
+fu_ccgx_hid_device_init (FuCcgxHidDevice *self)
+{
+	fu_device_set_protocol (FU_DEVICE (self), "com.cypress.ccgx");
+	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_REQUIRE_AC);
+//	fu_device_add_flag (FU_DEVICE (self), FWUPD_I2C_DEVICE_FLAG_UPDATABLE);
+	fu_hid_device_set_interface (FU_HID_DEVICE (self), 0x01);
```

Comment:
That's a little unusual; no?  Is this right?

Reply:
Unusual https://github.com/fwupd/fwupd/pull/1879/files#diff-91b79cd74a72843f7ec3c454bfd29b9aR27 -- but not unheard of https://github.com/fwupd/fwupd/blob/master/plugins/synaptics-cxaudio/fu-synaptics-cxaudio-common.h#L15